mm2src/gossipsub/src/protocol.rs

```diff
@@ -228,12 +230,25 @@ impl Decoder for GossipsubCodec {
                     "sequence number has an incorrect size",
                 ));
             }
+
+            let topics = publish.topic_ids.into_iter().map(TopicHash::from_raw);
```

Comment:
> I see the decode implementation if growing and it would be healthy to extract this

It's already defined as a inner-function for the exact reason. It's specifically created for `process_p2p_message`.

Reply:
@rozhkovdmitrii I see that the above code has changed, is your comment resolved yet? If not, can you please point to the new code you want fixed?